content/pytorch/concepts/using-functional-apis/using-functional-api.md

```diff
@@ -0,0 +1,79 @@
+---
+Title: 'Using Functional APIs'
+Description: 'Functional APIs in PyTorch provide a flexible way to define and manipulate neural networks using functions rather than object-oriented classes.'
+Subjects:
+  - 'Machine Learning'
+  - 'Data Science' 
+Tags:
+  - 'PyTorch'
+  - 'Deep Learning'
+  - 'Neural Networks'
+CatalogContent:
+  - 'learn-python'
+  - 'paths/data-science'
+---
+
+**Functional APIs** in PyTorch provide a flexible and powerful way to define and manipulate neural networks. Unlike the `torch.nn.Module` class, which uses an object-oriented approach, functional APIs allow you to define models using functions. This can be particularly useful for creating complex models, experimenting with new architectures, or when you need more control over the forward pass.
+
+## Key Concepts
+
+### Functional Layers
+
+Functional layers are stateless and do not store parameters. Instead, parameters are passed explicitly. This can be useful for creating custom layers or when you need to reuse the same layer with different parameters.
+
+Example:
+
+```python
+import torch
+import torch.nn.functional as F
+
+# Define input tensor
+x = torch.randn(10, 3)
+
+# Define weights and bias
+weight = torch.randn(5, 3, requires_grad=True)
+bias = torch.randn(5, requires_grad=True)
+
+# Apply linear transformation
+output = F.linear(x, weight, bias)
+```
+
+### Custom Loss Functions
+
+Using functional APIs, you can easily define custom loss functions. This is useful when the predefined loss functions in PyTorch do not meet your requirements.
+
+Example:
+```python
+def custom_loss(output, target):
+    loss = torch.mean((output - target) ** 2)
+    return loss
+
+# Example usage
+output = torch.randn(10, 5)
+target = torch.randn(10, 5)
+loss = custom_loss(output, target)
+```
+
+### Activation Functions
+
+PyTorch provides a variety of activation functions in the `torch.nn.functional` module. These can be used to add non-linearity to your models.
+
+Example:
+```python
+x = torch.randn(10, 5)
+
+# Apply ReLU activation
+output = F.relu(x)
+```
+
+## Advantages of Using Functional APIs
+
+1. **Flexibility**: Functional APIs provide more control over the forward pass and allow for easy experimentation with different architectures.
+2. **Reusability**: Since functional layers are stateless, they can be reused with different parameters without any side effects.
+3. **Customizability**: Easily define custom layers, loss functions, and activation functions to suit your specific needs.
+
+## Conclusion
+
+Functional APIs in PyTorch offer a powerful and flexible way to define and manipulate neural networks. By understanding and utilizing these APIs, you can create more complex and customized models, experiment with new architectures, and have greater control over the forward pass.
+
+For more information, refer to the [PyTorch documentation](https://pytorch.org/docs/stable/nn.functional.html).
```

Comment:
We can skip conclusion section, docs concept entries are not that long that we would need a conclusion section.

```suggestion
```